<launch>
  <arg name="color_octomap"           value="true"/>
  <arg name="map_resolution"          value="0.01"/>
  <param name="octomap_filename"      value="octomap"/>
  <param name="use_sim_time"          value="false"/>

  <node pkg="tf" type="static_transform_publisher" name="base_to_world"
        args="0.0 0.0 0.0 0.0 0.0 0.0 base_link world 100"/>

  <node name="lslidar_3d_slam" pkg="lslidar_3d_slam" type="lslidar_3d_slam" output="screen">
    <param name="base_frame"            type="string"     value="base_link"/>
    <param name="map_frame"             type="string"     value="map"/>
    <param name="publish_freq"          type="double"     value="1.0"/>
    <!--  点云话题  -->
    <param name="point_cloud_topic"     type="string"     value="color_points"/>
    <!--  匹配点云密度，数值越小，点云月稠密，特征越丰富，配准越准确，但占用更高的计算资源  -->
    <param name="voxel_grid_leaf_size_" type="double"     value="1.0"/>
    <!--  建图点云密度，数值越小，点云越稠密，但建图速度越慢，与车辆速度有关  -->
    <param name="min_map_distance"      type="double"     value="0.01"/>
    <!--  激光距离半径滤波，可以使地图看上去更加规整，但只有近距离点云很容易造成误匹配  -->
    <param name="min_scan_distance"     type="double"     value="5.0"/>
    <param name="max_scan_distance"     type="double"     value="30.0"/>
    <!--  是否开启地面拟合  -->
    <param name="enable_floor_filter"   type="bool"       value="true"/>
    <!--  是否开启闭环检测及优化  -->
    <param name="enable_loop_closeure"  type="bool"       value="true"/>
  </node>

  <group unless="$(arg color_octomap)">
    <node name="pcl_to_octomap" pkg="lslidar_3d_slam" type="pcl_to_octomap" output="screen">
      <!--  octomap地图保存时的点云分辨率  -->
      <param name="map_resolution"        type="double"     value="$(arg map_resolution)"/>
    </node>
  </group>

  <group if="$(arg color_octomap)">
    <node name="pcl_to_color_octomap" pkg="lslidar_3d_slam" type="pcl2color_octomap" output="screen">
      <!--  color_octomap地图保存时的点云分辨率  -->
      <param name="map_resolution"        type="double"     value="$(arg map_resolution)"/>
    </node>
  </group>

  <node name="rviz_node" pkg="rviz" type="rviz"
        args="-d $(find lslidar_3d_slam)/config/3d_slam.rviz" />

</launch>
